crates/ark/src/reticulate.rs

```diff
@@ -89,11 +98,13 @@ pub unsafe extern "C" fn ps_reticulate_open(input: SEXP) -> Result<SEXP, anyhow:
     let main = RMain::get();
 
     let input: RObject = input.try_into()?;
-    let input_code: Option<String> = input.try_into()?;
-
-    let mut comm_id_guard = RETICULATE_COMM_ID.lock().unwrap();
+    let input_code: Option<String> = match r_is_null(input.sexp) {
+        true => None,
+        false => Some(input.try_into()?),
```

Comment:
Use `if` instead of `match`? Seems a bit strange with a boolean. (Feel free to ignore)